src/tactic/suggest.lean

```diff
@@ -201,20 +192,19 @@ declare_trace suggest         -- Trace a list of all relevant lemmas
 -- count the number of local hypotheses used.
 private meta def apply_declaration_script
   (g : expr) (hyps : list expr)
-  (discharger : tactic unit := done)
+  (opt : opt := { })
   (d : decl_data) :
   tactic application :=
 -- (This tactic block is only executed when we evaluate the mllist,
 -- so we need to do the `focus1` here.)
 retrieve $ focus1 $ do
-  apply_declaration ff discharger d,
+  apply_declaration ff opt d,
   ng ← num_goals,
-  g ← instantiate_mvars g,
```

Comment:
Removing this was incorrect, and the test file clearly showed this. :-(

Without this, counting hypotheses used doesn't work, so the results are returned in a bad order. That's why I added those tests.

I've restored this, and added a comment explaining what it's here for.

Reply:
Ok. I removed it because `tactic_statement` calls that line as well and I thought it was redundant.